### PR TITLE
Create SteeringSystem.md

### PR DESCRIPTION
**Pull request by mamollica42**

### CODE REVIEW

File: Documentation/Signoffs/SteeringSystem.md
Comment (JesseTNRoberts) [inline Documentation/Signoffs/SteeringSystem.md]: Shouldn't this be no less?

File: Documentation/Signoffs/SteeringSystem.md
Comment (JesseTNRoberts) [inline Documentation/Signoffs/SteeringSystem.md:8]: I think there are further constraints/specs that are appropriate. Like not loose position when power cycled?

File: Documentation/Signoffs/SteeringSystem.md
Comment (JesseTNRoberts) [inline Documentation/Signoffs/SteeringSystem.md:66]: The purpose of this section is to analyze your specific design to show that it will meet the constraints. Remove any additional unnecessary details that aren't directly related to your design, specs, and analysis. Convince me this will do what you say that it needs to by analyzing appropriately.

**Comment by mamollica42** [comment]
The Steering Wheel System has been updated with the following changes:
1. Added Additional constraint regarding behavior when power is cycled.
2.  Added Wiring Schematic to show encoder connection to the board.
3.  Removed some unnecessary details in the analysis
4. Added some detail in the analysis that better describes how the constraints and specifications are being met

File: Documentation/Signoffs/SteeringSystem.md
Comment (JesseTNRoberts) [inline Documentation/Signoffs/SteeringSystem.md]: Again, shouldn't this be no less? Why are you requiring no more measurement than this?

File: Documentation/Signoffs/SteeringSystem.md
Comment (JesseTNRoberts) [inline Documentation/Signoffs/SteeringSystem.md:16]: This is not correct. There is no reason the measuring must be limited to this range.

File: Documentation/Signoffs/SteeringSystem.md
Comment (JesseTNRoberts) [inline Documentation/Signoffs/SteeringSystem.md]: This is not third party buildable. The signals being connected need to be enumerated and every connection shown. Also, I'm not sure how this picture was generated but it is not appropriate. Kicad, auto cad, etc are appropriate.

File: Documentation/Signoffs/SteeringSystem.md
Comment (JesseTNRoberts) [inline Documentation/Signoffs/SteeringSystem.md:24]: The steering wheel mechanical system has no way to mount apparently. Not buildable.

File: Documentation/Signoffs/SteeringSystem.md
Comment (JesseTNRoberts) [inline Documentation/Signoffs/SteeringSystem.md]: Encoder has nowhere to mount and does not have a way to connect to the gears. Not buildable.

File: Documentation/Signoffs/SteeringSystem.md
Comment (JesseTNRoberts) [inline Documentation/Signoffs/SteeringSystem.md]: Do not include this. The needed information should be in your schematic.

File: Documentation/Signoffs/SteeringSystem.md
Comment (JesseTNRoberts) [inline Documentation/Signoffs/SteeringSystem.md]: Do not include this. The needed information should be in your schematic.

File: Documentation/Signoffs/SteeringSystem.md
Comment (JesseTNRoberts) [inline Documentation/Signoffs/SteeringSystem.md]: Get rid of this. Constraints and functionality aren't introduced here.

File: Documentation/Signoffs/SteeringSystem.md
Comment (JesseTNRoberts) [inline Documentation/Signoffs/SteeringSystem.md]: This is not the purpose of this section.

File: Documentation/Signoffs/SteeringSystem.md
Comment (JesseTNRoberts) [reply Documentation/Signoffs/SteeringSystem.md:66]: This section is still bloated with unnecessary information. Prove your system will do what you said above and won't violate any constraints. Do it in an organized and explicit fashion.

Comment (JesseTNRoberts) [reply Documentation/Signoffs/SteeringSystem.md:66]: I see very little in the way of analysis here. Prove to me that it will work. That does not require paragraphs of writing. This section should be mostly numbers or simulation results. Include just enough writing so that what you are proving and how you proved it are clear.

File: Documentation/Signoffs/SteeringSystem.md
Comment (mamollica42) [reply Documentation/Signoffs/SteeringSystem.md]: The steering wheel is only able to turn 960 degrees total. If the measured angle was more than that, then we know the angle must be wrong. Therefore, the angle must be no more than 960 degrees from left to right.

File: Documentation/Signoffs/SteeringSystem.md
Comment (JesseTNRoberts) [inline Documentation/Signoffs/SteeringSystem.md]: How do you know that the micro and the encoder will communicate?

File: Documentation/Signoffs/SteeringSystem.md
Comment (JesseTNRoberts) [inline Documentation/Signoffs/SteeringSystem.md:77]: The resolution was not analyzed. I saw the Matlab script but it doesn't prove that the resolution is sufficient for your purpose. Probably that is because you may not have a firm idea of what the required resolution is. What resolution is needed for the application?

File: Documentation/Signoffs/SteeringSystem.md
Comment (JesseTNRoberts) [inline Documentation/Signoffs/SteeringSystem.md]: This does not sound like all the electrical considerations have been taken. Why is the current supplied by the pins of concern? The encoder will output to the microcontroller, not the other way around. Further, what about voltage?

Comment (mamollica42) [reply Documentation/Signoffs/SteeringSystem.md]: You are correct, I had it backwards when I first wrote it. I fixed it in the document. The Arduino pins can receive up to 40 mA and operate at 5 V. The Encoder outputs a maximum of 32 mA and the pins output 10-bit greycode.

File: Documentation/Signoffs/SteeringSystem.md
Comment (mamollica42) [reply Documentation/Signoffs/SteeringSystem.md]: They will communicate because the output of the encoder is simply 10-bit Grey code that can be read by the MCU by the digital pins. Additionally, the current and voltage ratings are such that the encoder will not damage the MCU.

Comment (JesseTNRoberts) [reply Documentation/Signoffs/SteeringSystem.md]: That does not mean that they will communicate. Bits have voltage associated and different devices operate at different voltages. You haven't considered the output voltage of the encoder as compared to the acceptable voltage levels for the inputs.

Gray code is not a communication method. It is an encoding.

File: Documentation/Signoffs/SteeringSystem.md
Comment (JesseTNRoberts) [reply Documentation/Signoffs/SteeringSystem.md]: Again grey code is not a method of communication. It is an encoding. Nothing (in the signoff) says that these will be able to communicate when you plug them up. "bits" don't exist electrically. They become voltages (usually). Different devices use different voltages to represent a 1 vs a 0. A "1" in some systems is any voltage greater than 1.5 v while in others it could be anything greater than 14v.

File: Documentation/Signoffs/SteeringSystem.md
Comment (JesseTNRoberts) [reply Documentation/Signoffs/SteeringSystem.md]: This does not logically track. The minimum rotation is 960 degrees. Why are you requiring no more than that? Further, you have not required that it have at least the minimum. 

If you needed to drive 200 miles and your car got 20 miles to the gallon, you need **at least** 10 gallons. However, it is not true that you must not have more than 10 gallons.

Your constraint is incorrect.

File: Documentation/Signoffs/SteeringSystem.md
Comment (JesseTNRoberts) [inline Documentation/Signoffs/SteeringSystem.md:13]: Not proven in the analysis section.

File: Documentation/Signoffs/SteeringSystem.md
Comment (JesseTNRoberts) [inline Documentation/Signoffs/SteeringSystem.md:14]: Not analyzed.

File: Documentation/Signoffs/SteeringSystem.md
Comment (JesseTNRoberts) [inline Documentation/Signoffs/SteeringSystem.md]: Within 1% suggests that you will have a precision of 0.1 degrees. However, you then later say the target precision is 9.6 degrees. That is very imprecise. 0.1 degrees seems an appropriate target.

File: Documentation/Signoffs/SteeringSystem.md
Comment (JesseTNRoberts) [inline Documentation/Signoffs/SteeringSystem.md:12]: My follow up question is why? Where does this constraint come from and why is 1 degree an appropriate constraint?

File: Documentation/Signoffs/SteeringSystem.md
Comment (JesseTNRoberts) [inline Documentation/Signoffs/SteeringSystem.md:139]: This design looks good. I'm ready to signoff. But, I am a bit concerned about the encoder selection. It is somewhat expensive. I recommend looking for a lower cost option (either a lower cost encoder or maybe a rotary potentiometer). However, I'll leave that as a decision for the team. Keep in mind that the total budget is 1000 dollars without specifically showing a unique need for additional funds. So, be careful to consider if the money may be needed somewhere else in the project.

**Review by JesseTNRoberts — APPROVED** [review]

**Comment by JesseTNRoberts** [comment]
I think I need to be added to have write access. I can't merge the pull request.

**Pull request merged**
